src/sandbox/profile/request/token.builder.js

```diff
@@ -0,0 +1,249 @@
+   *
+   * @returns {TokenRequestBuilder}
+   */
+  withGender(value, anchors = null) {
+    const sandboxAttribute = createAttribute(constants.ATTR_GENDER, value, anchors);
+    return this.withAttribute(sandboxAttribute);
+  }
+
+  /**
+   * @param {string} value
+   * @param {SandboxAnchor[]}
+   *
+   * @returns {TokenRequestBuilder}
+   */
+  withPhoneNumber(value, anchors = null) {
+    const sandboxAttribute = createAttribute(constants.ATTR_PHONE_NUMBER, value, anchors);
+    return this.withAttribute(sandboxAttribute);
+  }
+
+  /**
+   * @param {string} value
+   * @param {SandboxAnchor[]}
+   *
+   * @returns {TokenRequestBuilder}
+   */
+  withNationality(value, anchors = null) {
+    const sandboxAttribute = createAttribute(constants.ATTR_NATIONALITY, value, anchors);
+    return this.withAttribute(sandboxAttribute);
+  }
+
+  /**
+   * @param {string} value
+   * @param {SandboxAnchor[]}
+   *
+   * @returns {TokenRequestBuilder}
+   */
+  withPostalAddress(value, anchors = null) {
+    const sandboxAttribute = createAttribute(constants.ATTR_POSTAL_ADDRESS, value, anchors);
+    return this.withAttribute(sandboxAttribute);
+  }
+
+  /**
+   * @param {string} value
+   * @param {SandboxAnchor[]}
+   *
+   * @returns {TokenRequestBuilder}
+   */
+  withStructuredPostalAddress(value, anchors = null) {
+    const sandboxAttribute = createAttribute(
+      constants.ATTR_STRUCTURED_POSTAL_ADDRESS,
+      value,
+      anchors
+    );
+    return this.withAttribute(sandboxAttribute);
+  }
+
+  /**
+   * @param {string} value
+   * @param {SandboxAnchor[]}
+   *
+   * @returns {TokenRequestBuilder}
+   */
+  withSelfie(value, anchors = null) {
+    return this.withBase64Selfie(Buffer.from(value).toString('base64'), anchors);
+  }
+
+  /**
+   * @param {string} value
+   * @param {SandboxAnchor[]}
+   *
+   * @returns {TokenRequestBuilder}
+   */
+  withBase64Selfie(value, anchors = null) {
+    const sandboxAttribute = createAttribute(constants.ATTR_SELFIE, value, anchors);
+    return this.withAttribute(sandboxAttribute);
+  }
+
+  /**
+   * @param {string} value
+   * @param {SandboxAnchor[]}
+   *
+   * @returns {TokenRequestBuilder}
+   */
+  withEmailAddress(value, anchors = null) {
+    const sandboxAttribute = createAttribute(constants.ATTR_EMAIL_ADDRESS, value, anchors);
+    return this.withAttribute(sandboxAttribute);
+  }
+
+  /**
+   * @param {string} value
+   * @param {SandboxAnchor[]}
+   *
+   * @returns {TokenRequestBuilder}
+   */
+  withDocumentDetails(value, anchors = null) {
+    const sandboxAttribute = new SandboxAttributeBuilder()
+      .withName(constants.ATTR_DOCUMENT_DETAILS)
+      .withValue(value)
+      .withOptional(true)
+      .withAnchors(anchors)
+      .build();
+    return this.withAttribute(sandboxAttribute);
+  }
+
+  /**
+   * @returns {TokenRequest}
+   */
+  build() {
+    return new TokenRequest(
+      this.rememberMeId,
+      Object.keys(this.attributes).map(k => this.attributes[k])
```

Comment:
Would it not be better to store the attributes in an array originally?  By using a keyed dictionary, aren't we limiting the sandbox token to only allow one attribute with a certain name? (Also we wouldn't have to convert from a dictionary to array when building the request)

Reply:
_Java SDK_ uses a map similar to this - as discussed we can change to array/list when we add support for multiple attributes with the same name, but different constraints